src/include/uapi/ipc/control.h

```diff
@@ -152,4 +152,21 @@ struct sof_ipc_ctrl_data {
 	};
 } __attribute__((packed));
 
+/* event type */
+enum sof_ipc_ctrl_event_type {
+	SOF_CTRL_EVENT_KD_NOTIFICATION = 0,	/**< keyword detection event */
+};
+
+struct sof_ipc_comp_event {
+	struct sof_ipc_reply rhdr;
+	uint32_t src_comp_id;	/**< source component id */
```

Comment:
we should probably have the component type so that the notification can be understood unambiguously, e.g. the component tells you what the structure is all about and the component_id tells you what the source was. That would help should we ever have more that one detector.

Reply:
acknowledged